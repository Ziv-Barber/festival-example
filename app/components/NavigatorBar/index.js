// @flow

import React from 'react' // eslint-disable-line no-unused-vars
// import { withStyles } from '@material-ui/core/styles'
import { withRouter } from 'react-router-dom'
import styled from 'styled-components'

// Content that we need:
import mainNavLinks from '../../content/navigator'

// Material-ui stuff:
import MuiAppBar from '@material-ui/core/AppBar'
import MuiToolbar from '@material-ui/core/Toolbar'
import MuiTypography from '@material-ui/core/Typography'
import MuiButton from '@material-ui/core/Button'
import MuiMenuIcon from '@material-ui/icons/Menu'
import MuiMenu from '@material-ui/core/Menu'
import MuiMenuItem from '@material-ui/core/MenuItem'
import MuiIconButton from '@material-ui/core/IconButton'

import wings from './wings.png'

// Our types:
import type { NavigatorBarPropsType } from './index.types'

/**
 * NavigatorBar react component.
 * @reactProps {string} text Parameters for the component.
 */
const NavigatorBar = (props: NavigatorBarPropsType) => {
  const [menuEl, setMenuEl] = React.useState(null)
  const { className } = props

  const handleRoute = (routePath: string) => () => {
    props.history.push(routePath)
  }

  const handleMenu = (event) => {
    setMenuEl(event.currentTarget)
  }

  const handleMenuClose = () => {
    setMenuEl(null)
  }

  const handleMenuRoute = (routePath: string) => () => {
    handleRoute(routePath)()
    handleMenuClose()
  }

  const Menu = props.menuComp || MuiMenu

  return (
    <MuiAppBar className={className} variant="absolote">
      <MuiToolbar className="navbarRoot">
        <div className="navbarTitle">
          <MuiTypography
            variant="h6"
            color="inherit"
            noWrap
            onClick={handleRoute('/main/info')}
          >
            HeavenFest
          </MuiTypography>
        </div>
        {mainNavLinks.map((value) => (
          <MuiButton
            className="navbarButton"
            key={value.key}
            onClick={handleRoute(value.path)}
          >
            {value.display}
          </MuiButton>
        ))}
        <MuiIconButton
          className="navbarMenu"
          color="inherit"
          aria-owns={menuEl && 'menu-appbar'}
          aria-haspopup="true"
          onClick={handleMenu}
        >
          <MuiMenuIcon />
        </MuiIconButton>
        <Menu
          id="menu-appbar"
          className="navbarMenuRoot"
          classes={{
            paper: 'navbarMenuPaper'
          }}
          anchorEl={menuEl}
          anchorOrigin={{
            vertical: 'top',
            horizontal: 'right'
          }}
          transformOrigin={{
            vertical: 'top',
            horizontal: 'right'
          }}
          open={!!menuEl}
          onClose={handleMenuClose}
        >
          {mainNavLinks.map((value) => (
            <MuiMenuItem key={value.key} onClick={handleMenuRoute(value.path)}>
              {value.display}
            </MuiMenuItem>
          ))}
        </Menu>
      </MuiToolbar>
    </MuiAppBar>
  )
}

const RouteNavigatorBar = withRouter(NavigatorBar)

const StyledMenu = styled(MuiMenu)`
  & .navbarMenuPaper {
    background: ${(props) => props.theme.palette.background.default} !important;
  }
`

// $FlowFixMe
export default styled(RouteNavigatorBar).attrs({
  menuComp: StyledMenu
})`
  border-bottom: 1px #002966 solid;
  opacity: 0.9;

  & .navbarMenu {
  }

  & .navbarRoot {
    width: 100%;
    display: flex;
  }

  & .navbarTitle::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100px;
    height: 32px;
    background-image: url(${wings});
    background-repeat: no-repeat;
    background-size: 100px 32px;
  }

  & .navbarTitle {
    position: relative;
    cursor: pointer;
    padding-left: 108px;
    flex: 1;
    z-index: 1190;
    color: #eee0ff;
    text-shadow: 2px 2px #4d94ff;
  }

  & .navbarButton {
  }

  @media (max-width: 599px) {
    & .navbarButton {
      display: none;
    }
  }

  @media (min-width: 600px) {
    & .navbarMenu {
      display: none;
    }
  }
`
