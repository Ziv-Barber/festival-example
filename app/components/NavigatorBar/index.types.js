// @flow

export type NavigatorBarPropsType = {
  history: {
    push: (string) => void
  },
  className?: string
}
