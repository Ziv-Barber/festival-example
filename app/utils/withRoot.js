// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import { MuiThemeProvider } from '@material-ui/core/styles'
import { ThemeProvider } from 'styled-components'
import CssBaseline from '@material-ui/core/CssBaseline'
import theme from '../theme'

/**
 * withRoot HOC to wrap a react component with the app's theme provider.
 */
const withRoot = (Component: React.ComponentType<any>) => {
  const withRootBase = (props: any) => (
    <MuiThemeProvider theme={theme}>
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <CssBaseline />
          <Component {...props} />
        </React.Fragment>
      </ThemeProvider>
    </MuiThemeProvider>
  )

  // $FlowFixMe
  withRootBase.displayName = `withRoot(${Component.displayName ||
    Component.name})`

  return withRootBase
}

export default withRoot
