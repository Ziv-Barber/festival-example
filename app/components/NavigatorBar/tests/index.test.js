// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { render, cleanup } from 'react-testing-library'
import { BrowserRouter as Router } from 'react-router-dom'

import NavigatorBar from '../index'
import withRoot from '../../../utils/withRoot'

// For flow:
let { describe, it, afterEach, expect } = global

const TestComp = (prop) => (
  <Router>
    <NavigatorBar />
  </Router>
)

/** @test {NavigatorBar} */
describe('<NavigatorBar />', () => {
  afterEach(() => {
    cleanup()
  })

  /** @test {NavigatorBar} */
  it('should render', () => {
    const App = withRoot(TestComp)
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })
})
