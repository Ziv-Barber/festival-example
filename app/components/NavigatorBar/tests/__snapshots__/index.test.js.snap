// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigatorBar /> should render 1`] = `
<div>
  <header
    class="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 MuiAppBar-positionFixed-2 MuiAppBar-colorPrimary-8 mui-fixed NavigatorBar-sc-1dzex11-1 fFKGCF"
    variant="absolote"
  >
    <div
      class="MuiToolbar-root-37 MuiToolbar-regular-39 MuiToolbar-gutters-38 navbarRoot"
    >
      <div
        class="navbarTitle"
      >
        <h6
          class="MuiTypography-root-41 MuiTypography-h6-58 MuiTypography-colorInherit-70 MuiTypography-noWrap-67"
        >
          HeavenFest
        </h6>
      </div>
      <button
        class="MuiButtonBase-root-103 MuiButton-root-77 MuiButton-text-79 MuiButton-flat-82 navbarButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-78"
        >
          Lineup
        </span>
        <span
          class="MuiTouchRipple-root-125"
        />
      </button>
      <button
        class="MuiButtonBase-root-103 MuiButton-root-77 MuiButton-text-79 MuiButton-flat-82 navbarButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-78"
        >
          Tickets
        </span>
        <span
          class="MuiTouchRipple-root-125"
        />
      </button>
      <button
        class="MuiButtonBase-root-103 MuiButton-root-77 MuiButton-text-79 MuiButton-flat-82 navbarButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-78"
        >
          Travel
        </span>
        <span
          class="MuiTouchRipple-root-125"
        />
      </button>
      <button
        aria-haspopup="true"
        class="MuiButtonBase-root-103 MuiIconButton-root-106 MuiIconButton-colorInherit-107 navbarMenu"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label-111"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-112"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-125"
        />
      </button>
    </div>
  </header>
</div>
`;
